lib/OpenQA/Parser/Format/XUnit.pm

```diff
@@ -76,7 +79,8 @@ sub parse {
                 my $content = '# Test messages ';
                 $content .= "# $tc->{name}\n" if $tc->{name};
 
-                for my $out ($tc->children('skipped, passed, error, failure')->each) {
+                for my $out ($tc->children('skipped, passed, error, failure, softfailure')->each) {
+                    $tc_result = 'softfail' if ($out->tag =~ m/softfailure/);
```

Comment:
There's probably no need to use a regex here:
```suggestion
                    $tc_result = 'softfail' if $out->tag eq 'softfailure';
```
See also my other comment for the use of `$out->tag`.